tests/mir_test_framework/main.cpp

```diff
@@ -20,6 +20,7 @@
 
 #include <cstring>
 #include <iostream>
+#include <libgen.h>
```

Comment:
Does this compile anywhere?!

The Posix version of `basename()` takes a pointer to non-const: passing __FILE__ to it shouldn't compile.

We need something like:

    static char path[] = __FILE__;
    // Override this standard gtest message
    std::cout << "Running main() from " << basename(path) << std::endl;

Reply:
Without this include on musl:
```
[ 59%] Building CXX object tests/mir_test_framework/CMakeFiles/mir-public-test-framework.dir/main.cpp.o
/home/luca/mir/tests/mir_test_framework/main.cpp: In function 'int main(int, char**)':
/home/luca/mir/tests/mir_test_framework/main.cpp:28:44: error: 'basename' was not declared in this scope
     std::cout << "Running main() from " << basename(__FILE__) << std::endl;
                                            ^~~~~~~~
```
with this commit:
```
[ 59%] Building CXX object tests/mir_test_framework/CMakeFiles/mir-public-test-framework.dir/main.cpp.o
/home/luca/mir/tests/mir_test_framework/main.cpp: In function 'int main(int, char**)':
/home/luca/mir/tests/mir_test_framework/main.cpp:28:61: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
     std::cout << "Running main() from " << basename(__FILE__) << std::endl;
                                                             ^
```

At least it compiles :wink:

Reply:
Only if you ignore perfectly good diagnostics. 8^)